runtime/src/dice.rs

```diff
@@ -52,3 +53,31 @@ fn cert_from_dccm(dv: &DataVault, cert: &mut [u8], cert_type: CertType) -> Calip
 
     Ok(size)
 }
+
+/// Handle the get ldev cert message
+///
+/// Returns a TestGetCertResp and optionally the response size
+pub fn handle_get_ldevid_cert(dv: &DataVault) -> CaliptraResult<(TestGetCertResp, Option<usize>)> {
+    let mut resp = TestGetCertResp {
+        hdr: MailboxRespHeader::default(),
+        data: [0u8; 1024],
+    };
+
+    let cert_size = copy_ldevid_cert(dv, &mut resp.data)?;
+
+    Ok((resp, Some(cert_size + core::mem::size_of::<MailboxRespHeader>())))
+}
+
+/// Handle the get fmc alias cert message
+///
+/// Returns a TestGetCertResp and optionally the response size
+pub fn handle_get_fmc_alias_cert(dv: &DataVault) -> CaliptraResult<(TestGetCertResp, Option<usize>)> {
+    let mut resp = TestGetCertResp {
+        hdr: MailboxRespHeader::default(),
+        data: [0u8; 1024],
+    };
+
+    let cert_size = copy_fmc_alias_cert(dv, &mut resp.data)?;
+
+    Ok((resp, Some(cert_size + core::mem::size_of::<MailboxRespHeader>())))
```

Comment:
A couple small comments:

1. Could add size as a field to TestGetCertResp
2. Don't think the size needs to be optional

Reply:
Thanks, adding this size field let me clean things up a lot.